### DISCUSSION TITLE
please add documentation page for data prep

### DISCUSSION THREAD

**Discussion by 2x2xplz** [observablehq/plot#2221] [Ideas]
Trying to use Observable Plot in a web application (vanilla JS ES6). I'm searching and don't see any any documentation on how to prepare (or even import) a data file. For example, should JSON data be prepared as rows or columns?

```
const jsonRows = [
  {region: east, revenue: 400, profit: 100},
  {region: west, revenue: 250, profit: -75},
  {region: south, revenue: 500, profit: 60},
]

const jsonColumns = {
  region: [east, west, south],
  revenue: [400, 250, 500],
  profit: [100, -75, 60]
}
```

Then there all the edge cases: missing or n/a values, hierarchal data, mixed data types, optional values, etc.

Apologies if this is addressed and I missed it, but I'd suggest this is the first page after the "[getting started](https://observablehq.com/plot/getting-started)" docs page -- since you can't really get started if your data isn't ready to go.

**Comment by Fil** [comment]
see https://observablehq.com/plot/features/marks#marks-have-tidy-data

**Reply by mbostock** [reply]
It also is already covered in the first page after the Getting started guide.

https://observablehq.com/plot/features/plots#marks-option
